### DISCUSSION TITLE
VIP still assigned when vip-manager service is stopped

### DISCUSSION THREAD

**Discussion by sdayvz** [cybertec-postgresql/vip-manager#281] [Q&A]
Hi,
I use vip-manager 2.8.0 and consul 1.19.0
**When I stopped the vip-manager service, the VIP still assigned on server.** 
I can see the VIP like this:
```
2: ens224: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:50:56:b0:0a:a1 brd ff:ff:ff:ff:ff:ff
    altname enp19s0
    inet 10.10.10.10/24 brd 10.88.0.255 scope global noprefixroute ens224
       valid_lft forever preferred_lft forever
    inet 10.10.10.11/24 scope global secondary ens224
       valid_lft forever preferred_lft forever
``` 

Here is the vip-manager settings:

```

interval: 1000
trigger-key: "/patroni/dengage/leader"
trigger-value: "dengage-01"
ip: 10.10.10.11 
netmask: 24 
interface: ens224
hosting-type: basic 
dcs-type: consul
dcs-endpoints:
  - http://127.0.0.1:8500
consul-token: "f62e9800-282a-2baa-5564-71a79f87092e"
retry-num: 2
retry-after: 250  #in milliseconds
verbose: false

``` 

Is this a bug? If not, is there a setting to delete VIP automatically?
Thanks.

**Comment by pashagolub** [comment]
**UPD**: vip-manager removes VIP as the cleanup routine.

vip-manager doesn't remove VIP because there was no such a condition in consul. vip-manager does only one job: adds vip when it needed to be added, deletes vip when it should be deleted. 

In what way stopping the vip-manager affects the patroni cluster?

**Comment by sdayvz** [comment]
Hi @pashagolub ,
I just wanted to try adding and removing VIP to the server.
If I want to stop using vip-manager, I should delete VIP manually, right?

**Comment by pashagolub** [comment]
Well, yeah. Or make sure consul is in a state when VIP will be removed automatically

**Comment by pashagolub** [comment]
I've checked the code and I was wrong. As a cleanup routine vip-manager deconfigures virtual IP:https://github.com/cybertec-postgresql/vip-manager/blob/7ecae13f91c20c0bf89c1a04e66ac3d50f37f539/ipmanager/ip_manager.go#L58

Now the question is: why it doesn't in your case? Are there any additional information in logs? Crash maybe?

**Reply by balinorLiang** [reply]
When vip-manager exits, the deconfigureAddress function is supposed to run the cleanup routine, which executes the command:
`ip addr del 192.168.1.101/24 dev eth0`
However, this command fails to execute successfully. The log only shows got **error:** without any further details (no ExitError and output is empty).
Then @sdayvz and I faced with this issue, the VIP still assigned on server when vip-manager exited smoothly.
<img width="771" height="66" alt="image" src="https://github.com/user-attachments/assets/0f04a7a4-c2a5-40c5-a9f6-bcd71d35113f" />

**Comment by gaoxueyu** [comment]
I think this is a bug. Although it doesn't happen every time, I frequently encounter VIP unbinding failures in the scenario of gracefully stopping vip-manager—that is, the vip-manager has already stopped, but the VIP still remains. Do we have any plans to fix this issue in the future?

**Reply by pashagolub** [reply]
We would like to fix all the bugs. Do you have details how can we reproduce it?

**Reply by gaoxueyu** [reply]
> We would like to fix all the bugs. Do you have details how can we reproduce it?

I could reproduce this issue via #326. 

**First Scenario:**
Following are the detailed step:
```
[Unit]
Description=vip-manager
After=network.target
After=network-online.target
Wants=network-online.target

[Service]
Type=simple
User=root
Group=root
ExecStart=/usr/local/vip-manager/vip-manager --config=/usr/local/vip-manager/vip-manager.yml
Restart=on-failure
KillSignal=SIGINT

[Install]
WantedBy=multi-user.target
```

`systemctl stop vip-manager`


**second Scenario:**
Using kill -9 to smoothly shutdown could also reproduce.

Thanks,
Hope

**Comment by gaoxueyu** [comment]
Could you please provide a brief update on the status of this bug? If available, I’d also appreciate it if you could share the expected timeline for resolution.

**Reply by tpo** [reply]
FYI - it should be possible to contact [Cybertec ](https://www.cybertec-postgresql.com/en/) and place a commercial order with them to get this resolved if you really require to have this resolved at some specific date and can't do it yourself.
